commit 36b493d8ff9fc45dabe3bea17d922298a3950c8a
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Wed Mar 28 12:18:13 2018 +0100

    lan78xx: Ignore DT MAC address if already valid
    
    The patch to set the lan78xx MAC address from DT does so regardless of
    whether or not the interface already has a valid address. As the
    initialisation function is called from the reset handler when the
    interface is brought up, it is impossible to change the MAC address
    in a way that persists across the interface being brought up.
    
    Fix the problem by moving the DT reading code after the check for a
    valid address.
    
    See: https://www.raspberrypi.org/forums/viewtopic.php?f=28&t=209309
    
    Signed-off-by: Phil Elwell <phil@raspberrypi.org>

diff --git a/drivers/net/usb/lan78xx.c b/drivers/net/usb/lan78xx.c
index a2873f9bd851..0926ea652ebb 100644
--- a/drivers/net/usb/lan78xx.c
+++ b/drivers/net/usb/lan78xx.c
@@ -1643,14 +1643,6 @@ static void lan78xx_init_mac_address(struct lan78xx_net *dev)
 	u32 addr_lo, addr_hi;
 	int ret;
 	u8 addr[6];
-	const u8 *mac_addr;
-
-	/* maybe the boot loader passed the MAC address in devicetree */
-	mac_addr = of_get_mac_address(dev->udev->dev.of_node);
-	if (mac_addr) {
-		ether_addr_copy(addr, mac_addr);
-		goto set_mac_addr;
-	}
 
 	ret = lan78xx_read_reg(dev, RX_ADDRL, &addr_lo);
 	ret = lan78xx_read_reg(dev, RX_ADDRH, &addr_hi);
@@ -1663,6 +1655,15 @@ static void lan78xx_init_mac_address(struct lan78xx_net *dev)
 	addr[5] = (addr_hi >> 8) & 0xFF;
 
 	if (!is_valid_ether_addr(addr)) {
+		const u8 *mac_addr;
+
+		/* maybe the boot loader passed the MAC address in devicetree */
+		mac_addr = of_get_mac_address(dev->udev->dev.of_node);
+		if (mac_addr) {
+			ether_addr_copy(addr, mac_addr);
+			goto set_mac_addr;
+		}
+
 		/* reading mac address from EEPROM or OTP */
 		if ((lan78xx_read_eeprom(dev, EEPROM_MAC_OFFSET, ETH_ALEN,
 					 addr) == 0) ||
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1188_36b493d8ff9fc45dabe3bea17d922298a3950c8a.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1188_36b493d8ff9fc45dabe3bea17d922298a3950c8a.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
